godel_path_planning/src/trajectory_planning.cpp

```diff
@@ -0,0 +1,188 @@
+
+#include <tf_conversions/tf_eigen.h>
+
+
+// Anonymous namespace
+namespace
+{
+  const static double TOOL_POINT_DELAY = 0.75;
+
+  // Create a descartes RobotModel for graph planning
+  descartes_core::RobotModelPtr createRobotModel(const moveit::core::RobotStatePtr robot_state,
+                                                 const std::string& group_name,
+                                                 const std::string& tool_frame,
+                                                 const std::string& world_frame,
+                                                 const uint8_t iterations)
+  {
+    using descartes_core::RobotModelPtr;
+    using descartes_moveit::MoveitStateAdapter;
+
+    return RobotModelPtr(new MoveitStateAdapter(*robot_state, group_name, tool_frame, world_frame, iterations)); 
+  }
+
+  // Create a CartTrajectoryPt that defines a point & axis with free rotation about z
+  descartes_core::TrajectoryPtPtr pointAxisTrajectoryFactory(double x, double y, double z, 
+                                                             double rx, double ry, double rz,
+                                                             double angle_discretization)
+  {
+    using namespace descartes_core;
+    using namespace descartes_trajectory;
+
+    return boost::shared_ptr<TrajectoryPt>(
+      new CartTrajectoryPt(
+            TolerancedFrame(utils::toFrame(x,y, z, rx, ry, rz, descartes_core::utils::EulerConventions::XYZ),
+             ToleranceBase::zeroTolerance<PositionTolerance>(x, y, z),
+             ToleranceBase::createSymmetric<OrientationTolerance>(rx, ry, 0, 0, 0, 2.0 * M_PI)),
+            0.0, angle_discretization));
+  }
+
+  // Create an axial trajectory pt from a given tf transform
+  descartes_core::TrajectoryPtPtr tfToAxialTrajectoryPt(const tf::Transform& nominal, double discretization)
+  {
+    Eigen::Affine3d eigen_pose;
+    tf::poseTFToEigen(nominal,eigen_pose);
+    Eigen::Vector3d rpy =  eigen_pose.rotation().eulerAngles(0,1,2);
+
+    double rx = rpy(0);
+    double ry = rpy(1);
+    double rz = rpy(2);
+    double x = eigen_pose.translation()(0);
+    double y = eigen_pose.translation()(1);
+    double z = eigen_pose.translation()(2);
+
+    return pointAxisTrajectoryFactory(x, y, z , rx, ry, rz, discretization);
+  }
+
+  // Translates a point relative to a reference pose to an absolute transformation
+  // Also flips the z axis of the orientation to point INTO the plane specified by the
+  // reference frame.
+  tf::Transform createNominalTransform(const geometry_msgs::Pose& ref_pose, const geometry_msgs::Point& point)
+  {
+    // To plane
+    tf::Transform marker_pose;
+    tf::poseMsgToTF(ref_pose, marker_pose);
+    // From plane to point
+    tf::Transform to_point = tf::Transform::getIdentity();
+    to_point.setOrigin(tf::Vector3(point.x, point.y, point.z));
+    // Reverse orientation of z axis
+    tf::Quaternion quat;
+    quat.setEuler(0.0, 3.14159, 0.0); // yaw, pitch roll
```

Comment:
There is an 'M_PI' constant (in the math.h header I think) that you can use here.